src/transport_performance/gtfs/validation.py

```diff
@@ -601,3 +655,803 @@ def get_route_modes(self):
+            "eval_placeholder_1",
+            build_table(
+                validation_dataframe,
+                "green_dark",
+                padding="10px",
+                escape=False,
+            ),
+        )
+        eval_temp.insert("eval_title_1", "GTFS Feed Warnings and Errors")
+
+        eval_temp.insert(
+            "eval_placeholder_2",
+            build_table(self.feed.agency, "green_dark", padding="10px"),
+        )
+        eval_temp.insert("eval_title_2", "GTFS Agency Information")
+
+        eval_temp.insert(
+            "name_placeholder", self.feed.feed_info["feed_publisher_name"][0]
+        )
+        eval_temp.insert(
+            "url_placeholder",
+            self.feed.feed_info["feed_publisher_url"][0],
+            replace_multiple=True,
+        )
+        eval_temp.insert(
+            "lang_placeholder", self.feed.feed_info["feed_lang"][0]
+        )
+        eval_temp.insert(
+            "start_placeholder", self.feed.feed_info["feed_start_date"][0]
+        )
+        eval_temp.insert(
+            "end_placeholder", self.feed.feed_info["feed_end_date"][0]
+        )
+        eval_temp.insert(
+            "version_placeholder", self.feed.feed_info["feed_version"][0]
+        )
+
+        count_lookup = dict(self.feed.describe().to_numpy())
+        eval_temp.insert(
+            "agency_placeholder", str(len(count_lookup["agencies"]))
+        )
+        eval_temp.insert("routes_placeholder", str(count_lookup["num_routes"]))
+        eval_temp.insert("trips_placeholder", str(count_lookup["num_trips"]))
+        eval_temp.insert("stops_placeholder", str(count_lookup["num_stops"]))
+        eval_temp.insert("shapes_placeholder", str(count_lookup["num_shapes"]))
+
+        self.get_gtfs_files()
+        file_list_html = ""
+        for num, file in enumerate(self.file_list, start=1):
+            file_list_html = (
+                file_list_html
+                + f"""
+                    <div>
+                        <dd>{num}. </dd>
+                        <dt>{file}</dt>
+                    </div>"""
+            )
+
+        eval_temp.insert("eval_placeholder_3", file_list_html)
+        eval_temp.insert("eval_title_3", "GTFS Files Included")
+
+        eval_temp.insert("date", date)
+
+        with open(
+            f"{report_dir}/gtfs_report/index.html", "w", encoding="utf8"
+        ) as eval_f:
+            eval_f.writelines(eval_temp.get_template())
+
+        # stops
+        self.viz_stops(
+            out_pth=(
+                pathlib.Path(f"{report_dir}/gtfs_report/stop_locations.html")
+            )
+        )
+        self.viz_stops(
+            out_pth=pathlib.Path(f"{report_dir}/gtfs_report/convex_hull.html"),
+            geoms="hull",
+            geom_crs=27700,
+        )
+        stops_temp = TemplateHTML(
+            (
+                "src/transport_performance/gtfs/report/"
+                "html_templates/stops_template.html"
+            )
+        )
```

Comment:
another functional purity thing - lets pass it in as a parameter. No problem with these paths being set as default in the method signature though.

Reply:
same as above.